### PR TITLE
ContentStartPosition control and Examples new additions

### PR DESCRIPTION
**Pull request by fogbridgegit**
Overview
---
 Used to place the scene origin on startup
 
 A `ContentStartPosition` can be used to allow the user to move to a position and place content in that position.
 
 A `ContentStartPosition` covers a healthy portion of the screen and follows the users gaze(but doesnt screen lock).  It initially disables the content associated with it.
 It can be used to give important guidance to the user before allowing him/her to continue with the application.
 Once a user clicks on the `ContentStartPosition`, it will disappear and allow the user to view the scene.Additionally you can set a flag to re-position the scene object relative to where the `ContentStartPosition` was dismissed.  Some have used it as a type of splash screen before showing a initial set of menus or content.
 
 `ContentStartPosition` has three properties that can be set in the editor.
     bool MoveCollectionOnDismiss
     GameObject StartupObject
     float Distance 
     
 StartupObject refers to The scene object to activate and possibly reposition
 MoveCollectionOnDismiss toggles whether you want to reposition the StartupObject to the place where the Fitbox was clicked on (defaults to false) 
 Distance refers to the distance at which you want the `ContentStartPosition`, and subsequent content, to display in front of the user. This is fed to the interpolator. (defaults to 1.0f )
 
 `ContentStartPosition` requires an Interpolator, A Holotoolkit MonoBehaviour that interpolates a transform's position, rotation or scale.
 and helps the fitbox smoothly stay in front of the user wherever he/she moves
 
 A `ContentStartPosition` also has a child object Quad whose material can be set to a logo or some image representing guidance to the user
 You could replace this with your own child objects.
 
 If the user moves anywhere the fitbox will stay at the same distance set in front of the user and this is where the followup content may be moved to if MoveCollectionOnDismiss is set to true

### CODE REVIEW

File: Assets/HoloToolkit/Dialogs/Scripts/Fitbox.cs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+                    // Rotate the Hologram Collection to face the user.
+                    Quaternion toQuat = Camera.main.transform.localRotation * StartupObject.transform.rotation;
+                    toQuat.x = 0;
+                    toQuat.z = 0;
+                    StartupObject.transform.rotation = toQuat;
+                }
+            }
+            // Destroy the Fitbox
+            Destroy(gameObject);
```
Comment (StephenHodgson) [inline Assets/HoloToolkit/Dialogs/Scripts/Fitbox.cs]: Don't forget to add `eventData.Use();` to consume the input event.

Comment (fogbridgegit) [reply Assets/HoloToolkit/Dialogs/Scripts/Fitbox.cs]: ok just did.  do i need to do a new pull request or will this new commit just attach itself?

Comment (DDReaper) [reply Assets/HoloToolkit/Dialogs/Scripts/Fitbox.cs]: Just push your updates to your branch, that will update the PR automatically.

**Comment by DDReaper** [comment]
I think this would also be awesome for Immersive headsets, so long as it's not locked to using worldanchors.

Could it be use for immersive @fogbridgegit ?

**Comment by fogbridgegit** [comment]
no worries ill name it whatever you want.  let me know ill go back and text replace all references to fitbox

File: Assets/HoloToolkit/Dialogs/Scripts/Fitbox.cs (hunk 1 above)
Comment (fogbridgegit) [reply Assets/HoloToolkit/Dialogs/Scripts/Fitbox.cs]: done

**Comment by DDReaper** [comment]
I've put a request out in the MR_Toolkit channel on Slack to gauge peoples thoughts.

**Comment by fogbridgegit** [comment]
the example uses the MixedRealityCamera so probably can be used for immersive.  i Do have a Samsung IMD and the Emulator, maybe ill test it there in the coming days.

**Comment by DDReaper** [comment]
Any update @fogbridgegit on renaming the class in this PR and closing #1633

**Comment by StephenHodgson** [comment]
@fogbridgegit any update on this?

**Comment by HBJKZLH** [comment]
Untiy [Universal Windows Platform] Platform does not support.

The Socket code is written and the Thread code is written and the FileStream code is written, you know what?



I used the network communication section to find that this part of the code could not be transferred to the [Universal Windows Platform].

After I tested Unity 2017.2.0f3 (64-bit) version,Unity 2017.3.1f1 (64-bit) version,Unity 2017.2.1f1 (64-bit) version.



They don't support it.



Do you have any feedback on untiy?









When I wrote socket and IO and Thread related code in the program, untiy packaging times was wrong,
The socket.Connect function is not supported.
The socket.Receive function is not supported, and the 
socket.Send function is not supported.
The filestream.close function is not supported,
New Thread(recv) is not supported,
Does not support Thread. IsBackground,
Does not support Thread. To Start,




But if I just write in the code.

The Socket client = null;

Thread the Thread = null;

FileStream filedata = null;



This way, the Universal Windows Platform is passed.

I think it should be the flaw in the Platform library of untiy [Universal Windows Platform].


Please help me to feedback this question,








------------------ 原始邮件 ------------------
发件人: "Stephen Hodgson"<notifications@github.com>;
发送时间: 2018年3月2日(星期五) 凌晨1:37
收件人: "Microsoft/MixedRealityToolkit-Unity"<MixedRealityToolkit-Unity@noreply.github.com>;
抄送: "Subscribed"<subscribed@noreply.github.com>; 
主题: Re: [Microsoft/MixedRealityToolkit-Unity] ContentStartPositioncontrol and Examples new additions (#1623)




@fogbridgegit any update on this?

**Comment by StephenHodgson** [comment]
Closing stale PR

**Pull request closed without merging**